### PR TITLE
freeglut: update from 2.8.1 to 3.0.0

### PR DESCRIPTION
**Pull request by tehnick**

### CODE REVIEW

**Comment by tehnick** [comment]
Rebased on current master (again).

**Comment by pavelvat** [comment]
It is deprecated:
```
+    cd '$(1)/build' && cmake .. \
+        -DCMAKE_TOOLCHAIN_FILE='$(CMAKE_TOOLCHAIN_FILE)' \
```
see for example https://github.com/mxe/mxe/blob/master/src/miniupnpc.mk#L23-L24

**Comment by tehnick** [comment]
Ok. Thanks.

**Comment by pavelvat** [comment]
quotes for variable $(TARGET) are not required:
``` +    cd    '$(1).build' && '$(TARGET)-cmake' '$(1)' \ ```
use just:
``` +    cd    '$(1).build' && $(TARGET)-cmake '$(1)' \ ```

**Comment by pavelvat** [comment]
replace
```
+    mkdir '$(1)/build'
```
to
```
+    mkdir '$(1).build'
```

**Comment by starius** [comment]
Please separate build and install (install with -j1).

**Comment by tehnick** [comment]
Done.

**Comment by starius** [comment]
Tested on all 4 targets. Thanks!

File: src/freeglut.mk [hunk 1]
```diff
@@ -17,16 +17,19 @@ define $(PKG)_UPDATE
...
+        -DFREEGLUT_BUILD_SHARED_LIBS=$(if $(BUILD_STATIC),false,true)
+    $(MAKE) -C '$(1).build' -j '$(JOBS)'
+    $(MAKE) -C '$(1).build' -j 1 install
 
     '$(TARGET)-gcc' \
         -W -Wall -Werror -ansi -pedantic \
         '$(2).c' -o '$(PREFIX)/$(TARGET)/bin/test-freeglut.exe' \
-        `'$(TARGET)-pkg-config' glut --cflags --libs`
+        $(if $(BUILD_STATIC),-DFREEGLUT_STATIC) \
+        -L'$(PREFIX)/$(TARGET)/lib' -lglut -lglu32 -lopengl32 -lgdi32 -lwinmm
```
Comment (tonytheodore) [inline src/freeglut.mk]: `pkg-config` should be used for the test program to ensure the package can be used downstream. `cegui`, `libcaca`, `opencsg` appear to be broken by missing `-DFREEGLUT_STATIC` cflags.

Comment (tehnick) [inline src/freeglut.mk]: @tonytheodore
I see. As for now `freeglut` is cmake-based project, so `.pc` file should be generated manually.

Comment (tehnick) [inline src/freeglut.mk]: @tonytheodore 
Hmm, after quick look I see that `.pc` file is already generated by cmake. It just has wrong file name. I'll fix this ASAP.

Comment (tehnick) [inline src/freeglut.mk]: @tonytheodore 
#1429

**Pull request merged**